docs/arc42.md

```diff
@@ -0,0 +1,725 @@
+3. **Scalability**: Lightweight plugins ensure efficient handling of additional features and traffic.
+
+---
+
+## 6. Runtime View
+
+### 6.1 Message Flow
+1. Agent sends a message to the mediator.
+2. Mediator stores or forwards the message based on the recipient's status.
+3. Recipient retrieves the message via the pickup protocol.
+
+### Sequence Diagram
+```mermaid
+sequenceDiagram
+    Agent1->>Mediator: Send Message
+    Mediator->>Storage: Store Message
+    Agent2->>Mediator: Pickup Request
+    Mediator->>Agent2: Deliver Message
+```
+
+### 6.2 Mediator Coordination Flow
+The Mediation Coordination Protocol is a component of the DIDComm framework, facilitating secure and efficient message routing between agents. It enables a recipient agent to request a mediator agent to handle message forwarding on its behalf. 
+
+**Sequence Diagram: Mediation Coordination Protocol**
+
+```mermaid
+sequenceDiagram
+    participant Recipient
+    participant Mediator
+
+    Recipient->>Mediator: Mediate Request
+    alt Mediator Accepts
+        Mediator-->>Recipient: Mediate Grant
+        Note right of Mediator: Provides routing information
+    else Mediator Denies
+        Mediator-->>Recipient: Mediate Deny
+        Note right of Mediator: Declines mediation request
+    end
+    alt Recipient Registers Key
+        Recipient->>Mediator: Keylist Update
+        Mediator-->>Recipient: Keylist Update Response
+    end
+```
+
+**Illustration: Runtime Interaction**
+
+1. **Mediate Request**: The recipient sends a `mediate-request` message to the mediator, initiating the mediation process.
+
+2. **Mediator's Response**:
+   - **Grant**: If the mediator agrees, it responds with a `mediate-grant` message, providing necessary routing details.
+   - **Deny**: If the mediator declines, it sends a `mediate-deny` message.
+
+3. **Key Registration**: Upon receiving a grant, the recipient registers its keys with the mediator using a `keylist-update` message. The mediator acknowledges with a `keylist-update-response`.
+
+This interaction ensures that messages intended for the recipient are appropriately routed through the mediator, enhancing communication reliability and security.  
+
+---
+
+## 7. Deployment View
+
+### 7.1 Deployment Strategy
+
+The deployment of the `didcomm-mediator-rs` project leverages a scalable, distributed architecture designed for high availability, fault tolerance, and efficient message routing.
+
+---
+
+### 7.2 Deployment Models
+
+#### 1. **Single Instance Deployment (Minimal Setup)**
+
+**Use Case**: Suitable for development or testing environments.
+
+- A single mediator instance handles all requests and routing.
+- Simplifies deployment but lacks fault tolerance and scalability.
+
+**Advantages**:
+- Minimal resource usage.
+- Easy to configure and maintain.
+
+**Disadvantages**:
+- No fault tolerance: if the mediator fails, the service becomes unavailable.
+- Limited scalability.
+
+---
```

Comment:
remove this we are saying how the mediator is deployed not distinghuishing btw deployment models